usselman/alien-influencer-unity
# Building Destruction

# Major Update

## Folder Restructure
- Created folder /Assets/Base/ for internal assets, separate  from imported assets
- Migrated Scenes from /Library/ to /Assets/Base/Scenes

## Code Updates
- Minion and Building Class use an [enum based state controller](https://github.com/usselman/alien-influencer-unity/blob/ab161a096a1b976b46a9e4915f7d66c6c24834e0/Alien%20Influencer/Assets/Base/Scripts/Building.cs#L7), each state has a corresponding function. This allows:
- - Easy debugging for what is happening and why 
- - Clean distinction between actions that should happen once and actions that should be updated each frame
- New Singleton base class that allows inheritance with same behavior for each class : 
- - https://github.com/usselman/alien-influencer-unity/blob/ab161a096a1b976b46a9e4915f7d66c6c24834e0/Alien%20Influencer/Assets/Base/Scripts/BuildingHighlighter.cs#L5 
- - https://github.com/usselman/alien-influencer-unity/blob/ab161a096a1b976b46a9e4915f7d66c6c24834e0/Alien%20Influencer/Assets/Base/Scripts/UFOLaser.cs#L47

## Issues
This PR aims to solve #6 and #5 

## To Test 
- Open Assets/Base/Scenes/Level 1
- Highlight a Building by aiming the laser at it, select by pressing 'C'
- Confirm Minion
- - Walks to the Building
- - Attacks the Building
- - - Progress bar appears above building
- - - Progress bar increases in value with each hit
- - - Building explodes and rubble appears in its place once destroyed
- Minion walks to UFO when done attacking building
- - Progress on one building can be interrupted by attacking another building

## Merge Condition
@usselman approval or 10/2/24, whichever comes first
